packages/db/src/db.ts

```diff
@@ -0,0 +1,85 @@
+import { GraphQLSchema, DocumentNode, parse, execute } from "graphql";
+
+import { schema } from "@truffle/db/data";
+import { generateCompileLoad } from "@truffle/db/loaders/commands";
+import {
+  WorkflowCompileResult,
+  WorkspaceRequest
+} from "@truffle/db/loaders/types";
+import { Workspace } from "@truffle/db/workspace";
+
+interface IConfig {
+  contracts_build_directory: string;
+  contracts_directory: string;
+  working_directory?: string;
+  db?: {
+    adapter?: {
+      name: string;
+      settings?: any;
+    };
+  };
+}
+
+interface IContext {
+  artifactsDirectory: string;
+  workingDirectory: string;
+  contractsDirectory: string;
+  workspace: Workspace;
+  db: ITruffleDB;
+}
+
+interface ITruffleDB {
+  query: (query: DocumentNode | string, variables: any) => Promise<any>;
+}
+
+export class TruffleDB {
+  schema: GraphQLSchema;
+  context: IContext;
+
+  constructor(config: IConfig) {
+    this.context = this.createContext(config);
+    this.schema = schema;
+  }
+
+  async query(query: DocumentNode | string, variables: any = {}): Promise<any> {
+    const document: DocumentNode =
+      typeof query !== "string" ? query : parse(query);
+
+    return await execute(this.schema, document, null, this.context, variables);
+  }
+
+  async loadCompilations(result: WorkflowCompileResult) {
+    const saga = generateCompileLoad(result, {
+      directory: this.context.workingDirectory
+    });
+
+    let cur = saga.next();
```

Comment:
What does `cur` mean?

Reply:
It's the current spot at which the generator is paused. It's just a way to keep track and check whether the generator function is done or if there are more yields in there to get through. 

Each `saga.next()` call should hit the next `yield` in the generator function, and at each yield point in the generator the generator object is returned. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/function*